app/controllers/application_controller.rb

```diff
@@ -555,6 +555,20 @@ def search_clear
     end
   end
 
+  # Need this to be able to access controller methods in toolbar_builder when it is called from replace_right_cell method from controllers
+  # TODO: this will no longer be needed once all explorers are de-explorized
+  def controller
```

Comment:
@skateman i am not happy with this approach but did not know how else i can get this working, any advice will be good.
i think i have addressed all other comments/concerns, please review those as well.